app/Console/Commands/WaitingLists/SendAtcTopTenNotification.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace App\Console\Commands\WaitingLists;
+
+use App\Console\Commands\Command;
+use App\Models\Training\WaitingList;
+use App\Notifications\Training\WaitingListAtcTopTen;
+use Carbon\Carbon;
+
+class SendAtcTopTenNotification extends Command
+{
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'waitinglists:sendatctoptennotification';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Sends the top 10 members of an ATC waiting list a notification';
+
+    /**
+     * Executes all necessary console commands.
+     */
+    public function handle()
+    {
+        $waitingLists = WaitingList::all()->filter(function ($waitingList) {
+            return $waitingList->isAtcList();
+        });
+
+        foreach ($waitingLists as $waitingList) {
+            $nonNotifiedTopTenAccounts = $waitingList->accountsByEligibility(true)->take(10)->filter(function ($account) {
+                return is_null($account->pivot->within_top_ten_notification_sent_at);
+            });
+
+            foreach ($nonNotifiedTopTenAccounts as $account) {
+                $account->notify(new WaitingListAtcTopTen($waitingList->name));
+                $account->pivot->within_top_ten_notification_sent_at = Carbon::now();
+                $account->pivot->save();
+            }
+        }
+    }
```

Comment:
I would use collections here to avoid temporary variables, but that is a personal opinion - there is nothing functionally wrong with the code here.

Something like the below should work (although I've not tested it).

```php
        WaitingList::all()->filter(function (WaitingList $waitingList) {
            return $waitingList->isAtcList();
        })->mapWithKeys(function (WaitingList $waitingList) {
            return [$waitingList->name =>
                $waitingList
                    ->accountsByEligibility(true)
                    ->take(10)
                    ->filter(function ($account) {
                        return !!$account->pivot->within_top_ten_notification_sent_at;
                    })
            ];
        })->each(function (Collection $accounts, string $waitingListName) {
            $accounts->each(function ($account) use ($waitingListName) {
                $account->notify(new WaitingListAtcTopTen($waitingListName));
                $account->pivot->within_top_ten_notification_sent_at = Carbon::now();
                $account->pivot->save();
            });
        });
```